utils/markdown_test.go

```diff
@@ -0,0 +1,269 @@
+		},
+		{
+			name: "heading: ##### H5 header",
+			args: "##### H5 header",
+			want: "H5 header",
+		},
+		{
+			name: "heading: ###### H6 header",
+			args: "###### H6 header",
+			want: "H6 header",
+		},
+		{
+			name: "heading: multiline with header and paragraph",
+			args: "###### H6 header\nThis is next line.\nAnother line.",
+			want: "H6 header This is next line. Another line.",
+		},
+		{
+			name: "heading: multiline with header and list items",
+			args: "###### H6 header\n- list item 1\n- list item 2",
+			want: "H6 header list item 1 list item 2",
+		},
+		{
+			name: "heading: multiline with header and links",
+			args: "###### H6 header\n[link 1](https://mattermost.com) - [link 2](https://mattermost.com)",
+			want: "H6 header link 1 - link 2",
+		},
+		{
+			name: "list: 1. First ordered list item",
+			args: "1. First ordered list item",
+			want: "First ordered list item",
+		},
+		{
+			name: "list: 2. Another item",
+			args: "1. 2. Another item",
+			want: "Another item",
+		},
+		{
+			name: "list: * Unordered sub-list.",
+			args: "* Unordered sub-list.",
+			want: "Unordered sub-list.",
+		},
+		{
+			name: "list: - Or minuses",
+			args: "- Or minuses",
+			want: "Or minuses",
+		},
+		{
+			name: "list: + Or pluses",
+			args: "+ Or pluses",
+			want: "Or pluses",
+		},
+		{
+			name: "list: multiline",
+			args: "1. First ordered list item\n2. Another item",
+			want: "First ordered list item Another item",
+		},
+		{
+			name: "tablerow:)",
+			args: "Markdown | Less | Pretty\n" +
+				"--- | --- | ---\n" +
+				"*Still* | `renders` | **nicely**\n" +
+				"1 | 2 | 3\n",
+			want: "",
+		},
+		{
+			name: "table:",
+			args: "| Tables        | Are           | Cool  |\n" +
+				"| ------------- |:-------------:| -----:|\n" +
+				"| col 3 is      | right-aligned | $1600 |\n" +
+				"| col 2 is      | centered      |   $12 |\n" +
+				"| zebra stripes | are neat      |    $1 |\n",
+			want: "",
+		},
+		{
+			name: "strong: Bold with **asterisks** or __underscores__.",
+			args: "Bold with **asterisks** or __underscores__.",
+			want: "Bold with asterisks or underscores.",
+		},
+		{
+			name: "strong & em: Bold and italics with **asterisks and _underscores_**.",
+			args: "Bold and italics with **asterisks and _underscores_**.",
+			want: "Bold and italics with asterisks and underscores.",
+		},
+		{
+			name: "em: Italics with *asterisks* or _underscores_.",
+			args: "Italics with *asterisks* or _underscores_.",
+			want: "Italics with asterisks or underscores.",
+		},
+		{
+			name: "del: Strikethrough ~~strike this.~~",
+			args: "Strikethrough ~~strike this.~~",
+			want: "Strikethrough strike this.",
+		},
+		{
+			name: "links: [inline-style link](http://localhost:8065)",
+			args: "[inline-style link](http://localhost:8065)",
+			want: "inline-style link",
+		},
+		{
+			name: "image: ![image link](http://localhost:8065/image)",
+			args: "![image link](http://localhost:8065/image)",
+			want: "image link",
```

Comment:
Same for the image. The label provides absolutely no value to the reader. The URL, however, does. I'd prefer if we retained the URL in the converted string just like I proposed for the URL.